consensus/src/lib.rs

```diff
@@ -47,72 +49,17 @@ pub enum ConsensusError {
     FailedToReadProperty(BlockError),
 }
 
-impl ConsensusInterface {
-    pub fn subscribe_to_events(&mut self, handler: Arc<dyn Fn(ConsensusEvent) + Send + Sync>) {
-        self.consensus.subscribe_to_events(handler)
-    }
-
-    pub fn process_block(
-        &mut self,
-        block: Block,
-        source: BlockSource,
-    ) -> Result<(), ConsensusError> {
-        self.consensus
-            .process_block(block, source)
-            .map_err(ConsensusError::ProcessBlockError)?;
-        Ok(())
-    }
-
-    pub fn get_best_block_id(&self) -> Result<Id<Block>, ConsensusError> {
-        Ok(self
-            .consensus
-            .get_best_block_id()
-            .map_err(ConsensusError::FailedToReadProperty)?
-            .expect("There always must be a best block"))
-    }
-
-    pub fn is_block_in_main_chain(&self, block_id: &Id<Block>) -> Result<bool, ConsensusError> {
-        Ok(self
-            .consensus
-            .get_block_height_in_main_chain(block_id)
-            .map_err(ConsensusError::FailedToReadProperty)?
-            .is_some())
-    }
-
-    pub fn get_block_height_in_main_chain(
-        &self,
-        block_id: &Id<Block>,
-    ) -> Result<Option<BlockHeight>, ConsensusError> {
-        self.consensus
-            .get_block_height_in_main_chain(block_id)
-            .map_err(ConsensusError::FailedToReadProperty)
-    }
-
-    pub fn get_block_id_from_height(
-        &self,
-        height: &BlockHeight,
-    ) -> Result<Option<Id<Block>>, ConsensusError> {
-        self.consensus
-            .get_block_id_from_height(height)
-            .map_err(ConsensusError::FailedToReadProperty)
-    }
-
-    pub fn get_block(&self, block_id: Id<Block>) -> Result<Option<Block>, ConsensusError> {
-        self.consensus.get_block(block_id).map_err(ConsensusError::FailedToReadProperty)
-    }
-}
-
-impl subsystem::Subsystem for ConsensusInterface {}
+impl subsystem::Subsystem for Box<dyn ConsensusInterface> {}
```

Comment:
It would be nicer to have a type alias for a wrapper type `Box<dyn ConsensusInterface>`